workflow/rules/bwa-mem.smk

```diff
@@ -0,0 +1,48 @@
+# vim: syntax=python tabstop=4 expandtab
+# coding: utf-8
+
+__author__ = "Jonas Almlöf"
+__copyright__ = "Copyright 2021, Jonas Almlöf"
+__email__ = "jonas.almlof@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule bwa_mem:
+    input:
+        fastq=["Pre-alignment/merge/{sample}_R1.fq.gz", "Pre-alignment/merge/{sample}_R2.fq.gz"],
+    output:
+        bam="Alignment/bwa-mem/{sample}.bam",
+    log:
+        log="Logs/Aligment/bwa-mem/{sample}.log",
+    params:
+        index=config["references"]["genome"],
+        extra=config["bwa-mem"]["extra"],
+        sort=config["bwa-mem"].get("sort", "samtools")
+        sort_order=config["bwa-mem"].get("sort_order", "coordinate"),
+        sort_extra="-@ " + str(config["bwa-mem"]["threads"]),
+    threads: config["bwa-mem"]["threads"]
+    #benchmark:
+    #    repeat(_bwa_benchmark, config.get("benchmark", {}).get("repeats", 1))
+    container:
+        config["tools"].get("bwa-mem", config["tools"].get("default", ""))
+    message:
+        "{rule}: Align {wildcards.sample} with bwa and sort"
+    wrapper:
+        "0.70.0/bio/bwa/mem"
+
+
+rule index_bwa_bam:
+    input:
+        "Alignment/bwa-mem/{sample}.bam"
+    output:
+        "Alignment/bwa-mem/{sample}.bam.bai",
```

Comment:
I think the output should be in `index_bwa_bam` to have a link to the rule name or the log should be in `bwa-mem` but I guess there are certain problems with the naming conventions. I would also prefer if we use underscores everywhere and not a mixture to reduce confusion.

Reply:
Changed the bwa-mem to just bwa. 
In general I agree but in this case I don't. I guess if we make the indexing function more general this will solve the issue.

Reply:
I guess it would. Still we might need to find some standard of naming things. I am also concerned about the misc rules and the separate log directory. If we make them very unspecific such as:

```
input:
  {file}.bam
output:
  {file}.bai
```

The separate log directory might not work as the the `log` stanza needs to have the same wildcards I believe. It would be easier to do:

```
log:
  {file}_index.log
```

Reply:
`
output:
  {module}/{software}/{file}.bai
log:
   logs/{module}/{software}/{file}_index.txt
`